### PR TITLE
Adding Genetic Algorithm optimization functions to analytic_tests.tex

### LINKED ISSUE #1669: [TASK] Adding GA benchmark tests to analytic_tests.tex
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
Adding GA tests to analytic_tests.tex

**Describe the solution you'd like**
Just add the new tests to analytic_tests.tex

**Describe alternatives you've considered**
N/A

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by Jimmy-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1669

##### What are the significant changes in functionality due to this change request?
The optimization benchmark functions used in the Genetic algorithms will be documented

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: doc/tests/optimization_functions.tex [hunk 1]
```diff
@@ -117,7 +125,43 @@ \subsubsection{ND Slant}
...
+\subsubsection{Townsend}
+It is a highly nonlinear non convex function with multiple minima and maxima. The constraint is highly nonlinear and the y-domain is asymmetric.
+\begin{itemize}
+	\item Objective Function: $f(x,y) = -\left[\cos((x-0.1)y)\right]^2-x\sin(3x+y)$
+	\item Domain: $-2.25 \leq x \leq 2.25$ and $-2.5 \leq y \leq 1.75$
+	\item Constraint: $x^2 + y^2 \leq \left[2\cos(t)-\frac{1}{2}\cos(2t)-\frac{1}{4}\cos(3t) - \frac{1}{8}cos(4t) \right]^2+\left[2\sin(t)\right]^2 $;  \[ \text{where: } t=\arctan2(x,y) \]
+	\item Global Minimum: $f(2.0052938,1.1944509) = -2.0239884$
+\end{itemize}
+
+\subsubsection{Simionescu}
```
Comment (wangcj05) [inline doc/tests/optimization_functions.tex:160]: Do you have all the tests that corresponding to the objective functions that you added?

Comment (Jimmy-INL) [reply doc/tests/optimization_functions.tex:160]: I think one of them took forever to converge so I removed it. Also, I still need to add the discrete ones, and Diego has two or three that still need to be added.

Comment (wangcj05) [reply doc/tests/optimization_functions.tex:160]: If it is not used in the tests, I would suggest to comment it out in the document for now. @Jimmy-INL

**Comment by wangcj05** [comment]
@Jimmy-INL Anything else do you want to add to this PR?

**Review by mandd — DISMISSED** [review]

**Comment by aalfonsi** [comment]
Can you guys check the name of some options? (E.g. in the manual parentSelection can be Tournament (but in the code it is expected ```tournamentSelection```))

**Comment by Jimmy-INL** [comment]
Guys, I will fix what Andrea commented about in the manual here in this PR. I know it should be a separate issue and PR but we are in a hurry. Is that OK? In fact, I have already fixed them but will push here if that's OK.

**Comment by Jimmy-INL** [comment]
@aalfonsi are these the instances you meant? are there any other ones?

**Review by wangcj05 — APPROVED** [review]
PR looks good.

**Review by aalfonsi — APPROVED** [review]

**Comment by aalfonsi** [comment]
Thank you very much

**Pull request merged**